Makefile

```diff
@@ -36,36 +38,38 @@ publish: publish-pypi  ## Publish packages
 publish-pypi: clean package-pip lint-twine  ##- Publish Python packages to pypi
 	uv tool run twine upload dist/*
 
-.PHONY: setup
-setup: install-uv setup-precommit ## Set up a development environment
-	uv sync --frozen $(SETUP_TESTS_EXTRA_ARGS) --extra docs --extra lint --extra types
+# Find dependencies that need installing
+APT_PACKAGES :=
+ifeq ($(wildcard /usr/include/libxml2/libxml/xpath.h),)
+APT_PACKAGES += libxml2-dev
+endif
+ifeq ($(wildcard /usr/include/libxslt/xslt.h),)
+APT_PACKAGES += libxslt1-dev
+endif
+ifeq ($(wildcard /usr/share/doc/python3-venv/copyright),)
+APT_PACKAGES += python3-venv
+endif
+ifeq ($(wildcard /usr/share/doc/libyaml-dev/copyright),)
+APT_PACKAGES += libyaml-dev
+endif
+ifeq ($(wildcard /usr/share/doc/fuse-overlayfs/copyright),)
+APT_PACKAGES += fuse-overlayfs
+endif
+ifeq ($(wildcard /usr/share/doc/umoci/copyright),)
+APT_PACKAGES += umoci
+endif
 
 # Used for installing build dependencies in CI.
 .PHONY: install-build-deps
-install-build-deps: install-linux-build-deps install-lint-build-deps
-	# Ensure the system pip is new enough. If we get an error about breaking system packages, it is.
-	sudo pip install 'pip>=22.2' || true
-
-.PHONY: install-lint-build-deps
-install-lint-build-deps:
-ifeq ($(shell which apt-get),)
-	$(warning apt-get not found. Please install lint dependencies yourself.)
-else
-	sudo $(APT) install python-apt-dev libapt-pkg-dev clang
-endif
-
-.PHONY: install-linux-build-deps
-install-linux-build-deps:
-ifneq ($(OS),Linux)
+install-build-deps: install-lint-build-deps
+ifeq ($(APT_PACKAGES),)
 else ifeq ($(shell which apt-get),)
-	$(warning apt-get not found. Please install dependencies yourself.)
+	$(warning Cannot install build dependencies without apt.)
+	$(warning Please ensure the equivalents to these packages are installed: $(APT_PACKAGES))
 else
-	sudo $(APT) install libyaml-dev python3-dev python3-pip python3-setuptools \
-	  python3-venv python3-wheel fuse-overlayfs libapt-pkg-dev umoci libgit2-dev
-endif
-ifneq ($(shell which snap),)
-	sudo snap install lxd
```

Comment:
Installing `lxd` was also removed here. Is that on purpose?

Reply:
Yes; I don't know why that was there, there's already a target to pack the snap in starbase's makefile